core/src/main/java/org/pentaho/di/connections/ConnectionManager.java

```diff
@@ -58,14 +62,30 @@ public class ConnectionManager {
   private static final Logger logger = LoggerFactory.getLogger( ConnectionManager.class );
 
   private List<LookupFilter> lookupFilters = new ArrayList<>();
+
+  @NonNull
   private Supplier<IMetaStore> metaStoreSupplier;
+
+  @NonNull
+  private final Bowl bowl;
+
   private ConcurrentHashMap<String, ConnectionProvider<? extends ConnectionDetails>> connectionProviders =
     new ConcurrentHashMap<>();
 
   private Map<String, List<String>> namesByConnectionProvider = new ConcurrentHashMap<>();
   private Map<String, ConnectionDetails> detailsByName = new ConcurrentHashMap<>();
   private boolean initialized;
 
+  public ConnectionManager() {
+    // The default bowl also uses the default meta store supplier.
+    this( MetaStoreConst.getDefaultMetastoreSupplier(), DefaultBowl.getInstance() );
```

Comment:
Ugh. it's annoying that the Bowl exposes an `IMetaStore` and not a `Supplier<IMetaStore>`. This would be a bit cleaner if the Metastore came from the Bowl. 

I think that decision was mostly because the use cases that required a Supplier for deferred initialization didn't apply to Bowls, since once you have a Bowl, you're past the deferred initialization window. But, there are always exceptions like this. :( 

You could use `() -> DefaultBowl.getInstance().getMetastore()` and hope that the JIT elides away the lambda and the stack of Suppliers eventually. I believe we do that elsewhere.